deepchecks/nlp/context.py

```diff
@@ -289,8 +289,8 @@ def __init__(
             if test_dataset.has_label() and train_dataset.has_label() and not \
                     TextData.datasets_share_task_type(train_dataset, test_dataset):
                 raise DatasetValidationError('train_dataset and test_dataset must share the same label and task type')
-            if train_dataset.name == test_dataset.name:
-                raise DatasetValidationError('train_dataset and test_dataset must have different names')
+            # if train_dataset.name == test_dataset.name:
```

Comment:
Agree, you can just delete that instead of commenting out.

Reply:
On a second thought, I rely on the fact that dataset have different names to retrieve the appropriate prediction indexes from the prediction and proba vectors, so I do need the names not to be the same

Reply:
@noamzbr the name is None for both before the context assigns "Train" or "test" by default so they're the same at the point of the validation if a name is not specified.